pkg/module/blob/blob.go

```diff
@@ -0,0 +1,315 @@
+package blob
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	pathutil "path"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"github.com/opencontainers/go-digest"
+)
+
+// MediaTypeTar defines the media type for a tarred file
+const MediaTypeTar = "application/x-tar"
+
+// MediaTypeGZip defines the media type for a gzipped file
+const MediaTypeGZip = "application/gzip"
+
+// MediaTypeOctetStream is the media type for any binary data.
+const MediaTypeOctetStream = "application/octet-stream"
+
+// Output is the output if read BlobInput.
```

Comment:
I don't understand this sentence, but that might be due to my lack of background knowledge?

Reply:
It is true it is not very clear, I will change it to this:
`Output is the output generated when reading a blob.Input.`

Reply:
Excellent, thank you!